### PR TITLE
Use JUnit5 for Spring Boot Process Application

### PR DESCRIPTION
**Pull request by ingorichtsmeier**
Get rid of the old JUnit4 dependencies and use JUnit5 for new projects.

### CODE REVIEW

**Comment by tmetzke** [comment]
Really nice, @ingorichtsmeier 👍 
Tests look good as well with JUnit 5, I like this 🙂

**Review by tmetzke — CHANGES_REQUESTED** [review]
I like the changes 👍 
Can we manage the test coverage artifact version as a property in the root POM?

In general, should we update to JUnit 5 in all archetypes maybe? Is this something you could do? Not a pressing concern and we can do this as a follow-up as well, for sure.

File: camunda-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -90,18 +84,33 @@
...
-      <version>${version.camunda-test-coverage}</version>
+      <groupId>org.camunda.community.process_test_coverage</groupId>
+      <artifactId>camunda-process-test-coverage-junit5-platform-7</artifactId>
+      <version>2.0.0</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.camunda.community.process_test_coverage</groupId>
+      <artifactId>camunda-process-test-coverage-spring-test-platform-7</artifactId>
+      <version>2.0.0</version>
```
[the remark this answers is not in the document]
Comment (ingorichtsmeier) [reply camunda-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml:113]: There is already a version property for test coverage, set to 1.0.3. This is required for all other archetypes.

What do you prefer:

- [ ] a second version property 
- [ ] update all archetypes to the new process test coverage
- [ ] keep it and accept it until all archetypes are updated

I vote for the last option to get it merged.

[the remark this answers is not in the document]
Comment (tmetzke) [reply camunda-archetype-spring-boot/src/main/resources/archetype-resources/pom.xml:113]: Fair enough. Happy to go with option 3 for now 👍
I created a follow-up task to move to JUnit 5 in general with the archetypes: https://github.com/camunda/camunda-bpm-platform/issues/3157. Feel free to tackle that as well if you think that's helpful for users and have the resources to do so.

**Review by tmetzke — APPROVED** [review]

**Pull request merged**